packages/siop-oid4vp/lib/rp/RP.ts

```diff
@@ -417,7 +424,12 @@ export class RP {
       state,
       nonce,
       verification: mergeVerificationOpts(this._verifyResponseOptions, opts),
-      presentationDefinitions: opts?.presentationDefinitions ?? this._verifyResponseOptions.presentationDefinitions,
+      ...(opts?.presentationDefinitions && !opts?.dcqlQuery && {
+        presentationDefinitions: this._verifyResponseOptions.presentationDefinitions ?? opts?.presentationDefinitions
+      }),
+      ...(opts?.dcqlQuery && !opts?.presentationDefinitions && {
```

Comment:
Should do nothing with ! presentation definitions. It should check earlier whether both PD and DCQL are present and reject if that is the case as they are mutually exclusive

Reply:
Created the code to check whether PDs and DCQL are both present